pkg/fleet/internal/cdn/config.go

```diff
@@ -16,33 +19,85 @@ const (
 	doNotEditDisclaimer = `# This configuration was generated by Datadog's Fleet Automation. DO NOT EDIT.`
 )
 
-// config is a configuration for the package.
-type config map[interface{}]interface{}
-
 // layer is a config layer that can be merged with other layers into a config.
 type layer struct {
-	ID      string
-	Content map[interface{}]interface{}
+	ID                  string                 `json:"name"`
+	AgentConfig         map[string]interface{} `json:"config"`
+	SecurityAgentConfig map[string]interface{} `json:"security_agent"`
+	SystemProbeConfig   map[string]interface{} `json:"system_probe"`
```

Comment:
Just an FYI that I believe this is being updated soon to also support ClusterAgent configs. Not sure if the updater will be responsible for installing/configuring the cluster agent or not https://github.com/DataDog/datadog-operator/pull/1340/commits/51d2cf97370b4a5278df24135b119e4816e96415#diff-063b72ecc49939a1029c3d3820e85bf906aae54a38203c49563a0699fc0cce69R65

Reply:
Thanks for the FYI. That won't be necessary with the current state of things: everything k8s should be managed either by the operator or a new component. 

This installer is host-only